downstairs/src/region.rs

```diff
@@ -1285,6 +1316,20 @@ impl Extent {
         // Write any remaining data
         batched_pwritev.perform_writes()?;
 
+        // At this point, we know that the on-disk values for any written blocks
+        // must match the integrity hashes calculated above (and stored to
+        // SQLite).  We can therefore store hash values for many dirty blocks,
```

Comment:
Unfortunately, we don't know that until `flush` returns Ok - `write` returning ok is insufficient to know that bytes have landed in durable storage.

Reply:
After a few re-reads of this PR, this comment needs to change but at this point I think the PR itself is sound.

Reply:
Yeah, I was struggling for the right way to phrase "written-but-not-flushed" data, but "on-disk" is definitely misleading.

Changed in 485307332c91e99af6f35abf169475c6fb997954, let me know what you think of the new working.

Reply:
:+1: